### PR TITLE
[ISSUE-252] Expose duration of the different system utils

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
### Issue #252 

_Describe your changes_
* Add `SystemCMDDuration` metric and register it in prometheus in `init` function
* Evaluate duration of all system utils in methods in linuxutils package

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [x] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
```
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="0.025"} 0
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="0.05"} 0
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="0.1"} 0
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="0.5"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="1"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="3"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="5"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="10"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="15"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="30"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="45"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="90"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="180"} 3
system_utils_duration_seconds_bucket{method="LVRemove",name="/sbin/lvm lvremove --yes",le="+Inf"} 3
system_utils_duration_seconds_sum{method="LVRemove",name="/sbin/lvm lvremove --yes"} 0.741551443
system_utils_duration_seconds_count{method="LVRemove",name="/sbin/lvm lvremove --yes"} 3
```

### CODE REVIEW

File: pkg/base/linuxutils/fs/fs_helper.go [hunk 1]
```diff
@@ -102,10 +105,14 @@ func (h *WrapFSImpl) GetFSSpace(src string) (int64, error) {
...
 	*/
 
+	evalDuration := common.SystemCMDDuration.EvaluateDuration(prometheus.Labels{
+		"name":   strings.TrimSpace(fmt.Sprintf(CheckSpaceCmdImpl, "")),
+		"method": "GetFSSpace"})
 	stdout, _, err := h.e.RunCmd(fmt.Sprintf(CheckSpaceCmdImpl, src))
 	if err != nil {
 		return 0, err
 	}
+	evalDuration()
```
Comment (painhardcore) [inline pkg/base/linuxutils/fs/fs_helper.go:115]: if an error occurs - we are ignoring the duration?

Comment (AnastasiaShemyakinskaya) [reply pkg/base/linuxutils/fs/fs_helper.go:115]: I don't see the point of measuring duration of the failed command

Comment (painhardcore) [reply pkg/base/linuxutils/fs/fs_helper.go:115]: Why? Some operations can take time and eventually fail.
Also, besides duration, we also count the number of operations.

Comment (ykulazhenkov) [reply pkg/base/linuxutils/fs/fs_helper.go:115]: I agree with Andrey, I think we should collect metrics for failed calls also.

Comment (AnastasiaShemyakinskaya) [reply pkg/base/linuxutils/fs/fs_helper.go]: Fixed

File: pkg/base/command/cmdwithmetrics.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+// NewExecutorWithMetrics is a constructor for ExecutorWithMetrics
+func NewExecutorWithMetrics(exec CmdExecutor) *ExecutorWithMetrics {
+	return &ExecutorWithMetrics{CmdExecutor: exec}
+}
+
+// RunCmdWithMetrics runs RunCmd function with SystemCMDDuration metrics
+// Receive cmd interface for RunCmd, name of cmd and called method name for metric labels
+// Returns stdout, stderr and error in case of failed execution
+func (e *ExecutorWithMetrics) RunCmdWithMetrics(cmd interface{}, name, method string) (string, string, error) {
```
Comment (mishoyama) [inline pkg/base/command/cmdwithmetrics.go:39]: Do we plan to execute some command without collecting metrics?

File: pkg/metrics/common/sysutil.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/*
+Copyright © 2020 Dell Inc. or its subsidiaries. All Rights Reserved.
```
Comment (ykulazhenkov) [inline pkg/metrics/common/sysutil.go:2]: 2021 =)

**Review by ykulazhenkov — APPROVED** [review]

**Review by painhardcore — APPROVED** [review]

**Review by mishoyama — APPROVED** [review]

**Pull request merged**